wandb/sdk/launch/docker.py

```diff
@@ -45,6 +47,207 @@ def validate_docker_installation() -> None:
+        )
+    else:
+        py_version, py_major = get_current_python_version()
+
+    python_build_image = "python:{}".format(py_version)     # use bigger image for package installation
+    python_base_image = "python:{}-slim-buster".format(py_version)  # slim for running
+    if launch_project.gpu:
+        cuda_version = launch_project.cuda_version or "10.0"
+
+        # must install all python setup
+        if py_major == "2":
+            python_packages = [
+                "python{}".format(py_version),
+                "python-pip",
+                "python-setuptools",
+            ]
+        else:
+            python_packages = [
+                "python{}".format(py_version),
+                "python3-pip",
+                "python3-setuptools",
+            ]
+
+        base_setup = CUDA_SETUP_TEMPLATE.format(
+            python_packages=" \\\n".join(python_packages),
+            py_version=py_version,
+            cuda_base_image="nvidia/cuda:{}-runtime".format(cuda_version)
+        )
+    else:
+        python_packages = [
+            "python3-dev" if py_major == "3" else "python-dev",
+            "gcc",
+        ]  # required for python < 3.7
+
+        base_setup = PYTHON_SETUP_TEMPLATE.format(
+            py_base_image=python_base_image
+        )
+
+    username, userid = get_docker_user(launch_project)
+    workdir = "/home/{user}".format(user=username)
+
+    # add env vars
+    if _is_wandb_local_uri(api.settings("base_url")) and sys.platform == "darwin":
+        _, _, port = _, _, port = api.settings("base_url").split(":")
+        base_url = "http://host.docker.internal:{}".format(port)
+    elif _is_wandb_dev_uri(api.settings("base_url")):
+        base_url = "http://host.docker.internal:9002"
+    else:
+        base_url = api.settings("base_url")
+    env_vars_section = "\n".join(
+        [
+            f"ENV WANDB_BASE_URL={base_url}",
+            f"ENV WANDB_API_KEY={api.api_key}",
+            f"ENV WANDB_PROJECT={launch_project.target_project}",
+            f"ENV WANDB_ENTITY={launch_project.target_entity}",
+            f"ENV WANDB_LAUNCH={True}",
+            f"ENV WANDB_LAUNCH_CONFIG_PATH={os.path.join(workdir, DEFAULT_LAUNCH_METADATA_PATH)}",
+            f"ENV WANDB_RUN_ID={launch_project.run_id or None}",
+            f"ENV WANDB_DOCKER={launch_project.docker_image}",
+        ]
+    )
+
+    requirements_line = ""
+
+    if docker.is_buildx_installed():
+        requirements_line = "RUN --mount=type=cache,mode=0777,target={}/.cache,uid={},gid=0 ".format(  # todo: don't think this is working for partial caching
+            workdir, launch_project.docker_user_id
+        )
+    else:
+        wandb.termwarn(
+            "Docker BuildX is not installed, for faster builds upgrade docker: https://github.com/docker/buildx#installing"
+        )
+        requirements_line = "RUN WANDB_DISABLE_CACHE=true "
+    requirements_line += "pip install -r requirements.txt"
```

Comment:
Doesn't this drop the use of `_wandb_bootstrap`? What if the repo doesn't have a requirements.txt?

Reply:
fixed and readded this in